source/framework/core/inc/TRestEventProcess.h

```diff
@@ -167,13 +166,24 @@ class TRestEventProcess : public TRestMetadata {
 
    public:
     Int_t LoadSectionMetadata();
-    virtual void InitFromConfigFile() { ReadAllParameters(); }
+
+    virtual void InitFromConfigFile() {
+        std::map<string, string> params = GetParametersList();
+
+        for (auto it = params.begin(); it != params.end(); it++) {
+            it->second = fRunInfo->ReplaceMetadataMembers(it->second);
+            it->second = ReplaceMathematicalExpressions(it->second);
+        }
```

Comment:
I would suggest to move to structured bindings because it is more readable:

for (auto & [member,expresion] : params){
expresion = fRunInfo->ReplaceMetadataMembers(expresion);
expresion = ReplaceMathematicalExpressions(expresion);
}

Reply:
Hi @juanangp , thanks for these advices. It seems the code will not compile just as it is the cmake system configured.


```
/home/jgalan/rest-framework/source/framework/core/inc/TRestEventProcess.h:173:21: warning: decomposition declarations are a C++17 extension [-Wc++17-extensions]
```

Recently we moved from c++11 to c++14. Not sure what are the implications, issues or requirements to move to c++17. @nkx @Vindaar @lobis

Reply:
Yes, structured bindings are since C++17